lib/engine/src/trap/frame_info.rs

```diff
@@ -227,18 +237,22 @@ impl Drop for GlobalFrameInfoRegistration {
 /// dropped, will be used to unregister all name information from this map.
 pub fn register(
     module: Arc<ModuleInfo>,
-    finished_functions: &BoxedSlice<LocalFunctionIndex, FunctionBodyPtr>,
+    finished_functions: &BoxedSlice<LocalFunctionIndex, FunctionExtent>,
     frame_infos: PrimaryMap<LocalFunctionIndex, SerializableFunctionFrameInfo>,
 ) -> Option<GlobalFrameInfoRegistration> {
     let mut min = usize::max_value();
     let mut max = 0;
     let mut functions = BTreeMap::new();
-    for (i, allocated) in finished_functions.iter() {
-        let (start, end) = unsafe {
-            let ptr = (***allocated).as_ptr();
-            let len = (***allocated).len();
-            (ptr as usize, ptr as usize + len)
-        };
+    for (
+        i,
+        FunctionExtent {
+            ptr: start,
+            length: len,
+        },
+    ) in finished_functions.iter()
+    {
+        let start = **start as usize;
+        let end = start + len;
```

Comment:
Nice improvement!

I think the layout of `*const [T]` isn't guaranteed to be `ptr, len` so this is definitely an improvement in correctness